### PR TITLE
nydusify: remove dependency on docker to pull images when checking nydus images

### PR DESCRIPTION
**Pull request by wraymo**
#435 
Signed-off-by: wraymo <raywangv@gmail.com>

### CODE REVIEW

**Comment by jiangliu** [comment]
> /retest

Seems there's something issues related to nydus-snapshotter, which breaks our anolis-test suite, we are still solving the issue. Stay tuned:)

File: contrib/nydusify/pkg/checker/rule/filesystem.go [hunk 1]
```diff
@@ -162,25 +167,26 @@ func (rule *FilesystemRule) walk(rootfs string) (map[string]Node, error) {
...
-		Rootfs: rule.SourceMountPath,
-	}
-	if err := image.Pull(); err != nil {
-		return nil, errors.Wrap(err, "pull source image")
-	}
-	if err := image.Mount(); err != nil {
-		return nil, errors.Wrap(err, "mount source image")
+	layers := rule.SourceParsed.OCIImage.Manifest.Layers
+	for _, l := range layers {
+		reader, err := rule.SourceRemote.Pull(context.Background(), l, true)
```
Comment (imeoer) [inline contrib/nydusify/pkg/checker/rule/filesystem.go:179]: Maybe we need to make concurrency for layer pulling and limit the max concurrency.

File: contrib/nydusify/pkg/checker/rule/filesystem.go [hunk 2]
```diff
@@ -301,11 +307,9 @@ func (rule *FilesystemRule) Validate() error {
 		return nil
 	}
 
-	image, err := rule.mountSourceImage()
-	if err != nil {
+	if err := rule.mountSourceImage(); err != nil {
```
Comment (imeoer) [inline contrib/nydusify/pkg/checker/rule/filesystem.go:310]: Where is the overlay mount? and how do we clean up these directories after checking?

**Comment by jiangliu** [comment]
The CI issue has been solved, could you please help to rebase the PR?

**Comment by imeoer** [comment]
@wraymo Please take a look at the failed smoke testing: https://github.com/dragonflyoss/image-service/actions/runs/3095947889/jobs/5010927486#step:5:7200

```
mount: mounting overlay on tmp-4/nydus:image-from-2/fs/source_mounted failed: No such file or directory
```

**Comment by wraymo** [comment]
> @wraymo Please take a look at the failed smoke testing: https://github.com/dragonflyoss/image-service/actions/runs/3095947889/jobs/5010927486#step:5:7200
> 
> ```
> mount: mounting overlay on tmp-4/nydus:image-from-2/fs/source_mounted failed: No such file or directory
> ```

It's because overlayfs

> @wraymo Please take a look at the failed smoke testing: https://github.com/dragonflyoss/image-service/actions/runs/3095947889/jobs/5010927486#step:5:7200
> 
> ```
> mount: mounting overlay on tmp-4/nydus:image-from-2/fs/source_mounted failed: No such file or directory
> ```

fix it

**Comment by imeoer** [comment]
@wraymo Others LGTM! Last question, we should clean up the source directory once the check is finished.

**Review by imeoer — APPROVED** [review]

**Pull request merged**